### DISCUSSION TITLE
is it possible to make python ariadne on aws lambda?

### DISCUSSION THREAD

**Discussion by yeukfei02** [mirumee/ariadne#515] [Ideas]
I use `apollo-server-lambda` with `serverless-framework` and able to deploy aws lambda graphql, `ariadne` will be able to make on aws lambda too?

Thanks.

**Comment by kleysonr** [comment]
I dont know about AWS, but using Google Cloud Functions the code below works.

```
from ariadne import QueryType, graphql_sync, make_executable_schema, gql
from ariadne.constants import PLAYGROUND_HTML
from flask import Flask, request, jsonify

type_defs = gql("""
    type Query {
        hello: String!
    }
""")

query = QueryType()

@query.field("hello")
def resolve_hello(_, info):
    request = info.context
    user_agent = request.headers.get("User-Agent", "Guest")
    return "Hello, %s!" % user_agent


schema = make_executable_schema(type_defs, query)

def endpoint(request):
    
    if request.method == 'POST':
        
        # GraphQL queries are always sent as POST
        data = request.get_json()
        
        # Note: Passing the request to the context is optional.
        # In Flask, the current request is always accessible as flask.request
        success, result = graphql_sync(
            schema,
            data,
            context_value=request
        )

        status_code = 200 if success else 400
        return jsonify(result), status_code
    
    else:
        return PLAYGROUND_HTML, 200

```

**Reply by yeukfei02** [reply]
@kleysonr 

how do you deploy to google cloud function?
by using google cloud function cli or any other tools?
if you can share your project structure then would be great!

**Reply by kleysonr** [reply]
There are many ways to deploy a google cloud function. But using the web interface I believe be the easier way.

Try to follow this tutorial https://rominirani.com/google-cloud-functions-tutorial-writing-our-first-google-cloud-function-a62de60b5c90

Instead of node, select python 3.7 as the runtime.

In the code editor past the code above and for the `requirements.txt` use this:

```
# Function dependencies, for example:
# package>=version
ariadne
```
